### PR TITLE
Added streaming for InferenceClient

### PR DESCRIPTION
**Pull request by merveenoyan**
Added docs on streaming with InferenceClient.

### CODE REVIEW

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ client = InferenceClient(model=URL_TO_ENDPOINT_SERVING_TGI)
 client.text_generation(prompt="Write a code for snake game", model=URL_TO_ENDPOINT_SERVING_TGI)
 ```
 
+To stream tokens in `InferenceClient`, simply pass `stream=True`. Another parameter you can use with TGI backend is `details`. You can get more details on generation (tokens, probabilities, etc.) by `details` to `True`. By default, `details` is set to `False`, and `text_generation` only returns text output. If you set `details` and `stream` both as `True`, `text_generation` will return `TextGenerationStreamResponse` which consists of the generated token, generated text, and details.
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:36]: Maybe add an example snippet below this with example outputs 🔥 cool stuff!

Comment (Wauplin) [inline docs/source/basic_tutorials/consuming_tgi.md:36]:
```suggestion
To stream tokens in `InferenceClient`, simply pass `stream=True`. Another parameter you can use with TGI backend is `details`. You can get more details on generation (tokens, probabilities, etc.) by setting `details` to `True`. By default, `details` is set to `False`, and `text_generation` returns a string. If you pass `details=True` and `stream=True`, `text_generation` will return a `TextGenerationStreamResponse` which consists of the generated token, generated text, and details.
```

Suggesting a few tweaks. Thanks @merveenoyan for working on these docs!

Comment (osanseviero) [reply docs/source/basic_tutorials/consuming_tgi.md]: @merveenoyan this is still pending

**Comment by merveenoyan** [comment]
@osanseviero sorry for delay, you can see examples now (forgot to commit :D)

File: docs/source/basic_tutorials/consuming_tgi.md [hunk 2]
```diff
@@ -33,6 +33,26 @@ client = InferenceClient(model=URL_TO_ENDPOINT_SERVING_TGI)
...
+
+```python
+output = client.text_generation(prompt="Meaning of life is", model="http://localhost:3000/", stream=True, details=True)
+print(next(iter(output)))
+
+# TextGenerationStreamResponse(token=Token(id=267, text=' a', logprob=-2.0723474, special=False), generated_text=None, details=None)
+```
+
+
+
```
Comment (osanseviero) [inline docs/source/basic_tutorials/consuming_tgi.md:55]:
```suggestion
```

**Review by osanseviero — DISMISSED** [review]
Thanks!

**Review by osanseviero — APPROVED** [review]
LGTM!

**Pull request merged**